lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -155,6 +155,7 @@ def apply_callback(data):
 
         # Redraw
         figure = axes.get_figure()
+        axes.legend()
```

Comment:
The point of the block of code above this is to only do this if the user has asked.  This will unconditionally add a legend, even if the user did not have one before and will destroy any non-default legend the user has created.